### ISSUE TITLE
Annotations for human36 3D pose estimation

### ISSUE THREAD

**Issue by julianzille** [open-mmlab/mmpose#1679]
Hi everyone. 

I am planning on using my own annotated dataset to perform 3D pose estimation leveraging the config and dataset structure of the _fully_ supervised Human 3.6 model `configs/body/3d_kpt_sview_rgb_vid/video_pose_lift/h36m/videopose3d_h36m_27frames_fullconv_supervised.py` ([config](https://github.com/open-mmlab/mmpose/blob/master/configs/body/3d_kpt_sview_rgb_vid/video_pose_lift/h36m/videopose3d_h36m_27frames_fullconv_supervised.py)). 

As far as I know, it only requires 2D annotated frames (x,y coords only) of various motion sequences to train and validate. Is this correct? In `tools/dataset/preprocess_h36m.py`, the 2D _and_ 3D coordinates are extracted from the original human 3.6 annotation file, but that does not necessarily mean that the 3D coordinates are used in training/validation (ie. only the 2D annotations are used). 

This is something I just want to confirm! Many thanks in advance :)

**Comment by ly015** [comment]
3D keypoint coordinates are necessary for model training and testing. On the contrary, I think 2D annotations are not necessary since they can be obtained with 3D coordinates and camera parameters.

**Comment by julianzille** [comment]
Thank you @ly015. However, the [paper](https://arxiv.org/abs/1811.11742) seems to suggest that only 2D labelled data and camera intrinsic/extrinsic parameters are required...

"Our temporal convolutional model takes 2D keypoint sequences (bottom) as input and generates 3D pose estimates as output (top)"

"... collecting labels for 3D human pose estimation requires an expensive motion capture setup as well as lengthy recording sessions."

"First, we present a simple and efficient approach for 3D human pose estimation in video _based on dilated temporal convolutions on 2D keypoint trajectories_"

"The input layer takes the concatenated (x, y) coordinates of the J joints for each frame and applies a temporal convolution with kernel size W and C output channels."

The [config file](https://github.com/open-mmlab/mmpose/blob/master/configs/body/3d_kpt_sview_rgb_vid/video_pose_lift/h36m/videopose3d_h36m_27frames_fullconv_supervised.py) also suggests that only 2D keypoints are required:
```
model = dict(
    type='PoseLifter',
    pretrained=None,
    backbone=dict(
        type='TCN',
        in_channels=2 * 17,
```
My feeling is that the `preprocess_h36.py` file extracts the 3D keypoints from the original dataset, but they are not used, and only the 2D keypoints actually get used.

**Comment by ly015** [comment]
1. This paper introduced a pose lifting model that lifts 2D coordinates to 3D. To train the model, 2D coordinates are fed as the input, and 3D ground-truth coordinates are used as the supervision of the model output.
2. This paper further proposed a semi-supervised train approach, that leverage **extra** unlabeled 2D data (only 2D keypoint coordinates, without 3D ground-truth) to boost the model performance. A certain amount of data with 3D coordinates is still needed.

**Comment by ly015** [comment]
You can use the pre-trained model on h36m to inference 3D keypoints from your own 2D data, although the accuracy may be poor due to the difference in camera parameters and data distribution.

If you want to train models on your own data, 3D keypoint coordinates are necessary. This paper introduced a pose lifting model that lifts 2D coordinates to 3D. To train the model, 2D coordinates are fed as the input, and 3D ground-truth coordinates are used as the supervision of the model output.
This paper further proposed a semi-supervised train approach, that leverage extra unlabeled 2D data (only 2D keypoint coordinates, without 3D ground-truth) to boost the model performance. A certain amount of data with 3D coordinates is still needed.

**Comment by julianzille** [comment]
Thank you. That helps a lot!
